libraries/core/lib/search/objective/BranchObjectiveFunction.ts

```diff
@@ -31,32 +30,27 @@ import { ApproachLevel } from "./ApproachLevel";
  * @author Annibale Panichella
  * @author Dimitri Stallenberg
  */
-export class BranchObjectiveFunction<T extends Encoding>
-  implements ObjectiveFunction<T>
-{
+export class BranchObjectiveFunction<
+  T extends Encoding
+> extends ObjectiveFunction<T> {
   protected _subject: SearchSubject<T>;
   protected _id: string;
-  protected _line: number;
-  protected _type: boolean;
 
   /**
    * Constructor.
    *
    * @param subject
    * @param id
-   * @param line
-   * @param type
    */
   constructor(
+    approachLevel: ApproachLevel,
+    branchDistance: BranchDistance,
     subject: SearchSubject<T>,
-    id: string,
-    line: number,
-    type: boolean
+    id: string
   ) {
+    super(approachLevel, branchDistance);
```

Comment:
For approachlevel this would work. But the branchdistance is language specific as it depends on the AST types. For this reason it cannot be static.

Another solution would be to create our own format for calculating the branchdistance. (but this will be a lot of work and hard to maintain)

We could create a subclass but the plugins would still always require them.

Reply:
Done.